### PR TITLE
Add command line options

### PR DESCRIPTION
**Pull request by spicylemonhaha**
# The main changes are as follows
1. Added command line options for start and build
2. Now, users can use their own declared configuration file

### CODE REVIEW

File: docs/zh/api/config-extension.md [hunk 1]
```diff
@@ -22,6 +22,13 @@ export default defineConfig({
 });
 ```
 
+当然你也可以通过命令行选项来添加 vite 配置，但是仅限于` island build docs`与` island dev docs`命令（我们的构建预览并没有使用 vite），目前支持的命令行选项可以通过`--help`命令来获取：
```
Comment (sanyuan0704) [inline docs/zh/api/config-extension.md:25]: There is useless space.

File: src/node/build.ts [hunk 2]
```diff
@@ -60,13 +61,15 @@ interface ServerEntryExports {
 class SSGBuilder {
   #root: string;
   #config: SiteConfig<unknown>;
+  #options: BuildOption;
```
Comment (sanyuan0704) [inline src/node/build.ts:64]: It is better to call it `CLIBuildOptions`.

File: src/node/serve.ts [hunk 3]
```diff
@@ -9,14 +9,15 @@ export interface ServeOptions {
...
   root?: string;
   port?: number;
+  host?: string;
 }
 
 // Serve ssg site in production
-export async function serve(root: string, userPort: number) {
-  const port = userPort !== undefined ? userPort : 4173;
+export async function serve(root: string, options: ServeOptions) {
+  const port = options.port !== undefined ? options.port : 4173;
```
Comment (sanyuan0704) [inline src/node/serve.ts:17]: const port = options.port ?? 4173;

File: src/node/serve.ts [hunk 4]
```diff
@@ -9,14 +9,15 @@ export interface ServeOptions {
...
+  const host = options.host !== undefined ? options.host : 'localhost';
```
Comment (sanyuan0704) [inline src/node/serve.ts:18]: The same with above.

File: src/node/config.ts [hunk 5]
```diff
@@ -18,18 +19,53 @@ type RawConfig<ThemeConfig = unknown> =
   | Promise<UserConfig<ThemeConfig>>
   | (() => UserConfig<ThemeConfig> | Promise<UserConfig<ThemeConfig>>);
 
+/**
```
Comment (sanyuan0704) [inline src/node/config.ts:22]: If there is single line commemt, you could use `//` instead.

File: src/node/config.ts [hunk 6]
```diff
@@ -18,18 +19,53 @@ type RawConfig<ThemeConfig = unknown> =
...
+ * resolve .island files
+ */
 const resolve = (root: string, ...files: string[]) =>
   path.resolve(root, '.island', ...files);
 
+/**
+ * get get defaultConfigName or userConfigName
```
Comment (sanyuan0704) [inline src/node/config.ts:29]: Get default config file or user custom config file.

File: src/node/config.ts [hunk 7]
```diff
@@ -18,18 +19,53 @@ type RawConfig<ThemeConfig = unknown> =
...
+const getConfigFileName = (customizePath?: string) => {
+  if (customizePath) {
+    const [path] = customizePath.split('.');
+    return path;
+  } else {
+    return 'config';
+  }
+};
+/**
+ * get defaultConfigPath or userConfigPath
```
Comment (sanyuan0704) [inline src/node/config.ts:40]: Remove this becuase it's not neccessary.

File: src/node/config.ts (hunk 5 above)
Comment (spicylemonhaha) [reply src/node/config.ts:22]: When someone wants to see the description of the function through hover, `/**` can help him。
image

File: docs/zh/guide/configure-site.md [hunk 8]
```diff
@@ -25,7 +25,9 @@ export default defineConfig({
 });
 ```
 
-关于配置文件，有以下两个注意事项:
+你也可以显式地通过 --config 命令行选项指定一个配置文件（配置文件仍然需要放在.island路径下）。
```
Comment (sanyuan0704) [inline docs/zh/guide/configure-site.md:28]: There should be space around.

**Review by sanyuan0704 — APPROVED** [review]

**Review by sanyuan0704 — APPROVED** [review]

**Pull request merged**